### PR TITLE
[Object bricks] Add events when changing object brick definition

### PR DESCRIPTION
**Pull request by BlackbitDevs**
When a data object class gets created, saved or deleted there are events like `pimcore.class.preAdd`, `pimcore.class.postUpdate` or `pimcore.class.postDelete`, see https://github.com/pimcore/pimcore/blob/11.x/lib/Event/DataObjectClassDefinitionEvents.php. This PR adds similar events for object bricks: 
- `pimcore.objectbrick.preAdd`
- `pimcore.objectbrick.postAdd`
- `pimcore.objectbrick.preUpdate`
- `pimcore.objectbrick.postUpdate`
- `pimcore.objectbrick.preDelete`
- `pimcore.objectbrick.postDelete`

### CODE REVIEW

File: lib/Event/ObjectbrickDefinitionEvents.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     * @var string
+     */
+    const PRE_DELETE = 'pimcore.objectbrick.preDelete';
+
+    /**
+     * @Event("Pimcore\Event\Model\DataObject\ClassDefinitionEvent")
+     *
+     * @var string
+     */
+    const POST_DELETE = 'pimcore.objectbrick.postDelete';
```
Comment (jdreesen) [inline lib/Event/ObjectbrickDefinitionEvents.php]:
```suggestion
    /**
     * @Event("Pimcore\Event\Model\DataObject\ObjectbrickDefinitionEvent")
     *
     * @var string
     */
    const PRE_ADD = 'pimcore.objectbrick.preAdd';

    /**
     * @Event("Pimcore\Event\Model\DataObject\ObjectbrickDefinitionEvent")
     *
     * @var string
     */
    const POST_ADD = 'pimcore.objectbrick.postAdd';

    /**
     * @Event("Pimcore\Event\Model\DataObject\ObjectbrickDefinitionEvent")
     *
     * @var string
     */
    const PRE_UPDATE = 'pimcore.objectbrick.preUpdate';

    /**
     * @Event("Pimcore\Event\Model\DataObject\ObjectbrickDefinitionEvent")
     *
     * @var string
     */
    const POST_UPDATE = 'pimcore.objectbrick.postUpdate';

    /**
     * @Event("Pimcore\Event\Model\DataObject\ObjectbrickDefinitionEvent")
     *
     * @var string
     */
    const PRE_DELETE = 'pimcore.objectbrick.preDelete';

    /**
     * @Event("Pimcore\Event\Model\DataObject\ObjectbrickDefinitionEvent")
     *
     * @var string
     */
    const POST_DELETE = 'pimcore.objectbrick.postDelete';
```

File: lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return $this->classDefinition;
+    }
+
+    /**
+     * @param Definition $classDefinition
+     */
+    public function setClassDefinition(Definition $classDefinition): void
+    {
+        $this->classDefinition = $classDefinition;
+    }
```
Comment (jdreesen) [inline lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php]:
```suggestion
    protected Definition $objectbrickDefinition;

    /**
     * @param Definition $objectbrickDefinition
     */
    public function __construct(Definition $objectbrickDefinition)
    {
        $this->objectbrickDefinition = $objectbrickDefinition;
    }

    /**
     * @return Definition
     */
    public function getObjectbrickDefinition(): Definition
    {
        return $this->objectbrickDefinition;
    }

    /**
     * @param Definition $objectbrickDefinition
     */
    public function setObjectbrickDefinition(Definition $objectbrickDefinition): void
    {
        $this->objectbrickDefinition = $objectbrickDefinition;
    }
```

File: models/DataObject/Objectbrick/Definition.php [hunk 3]
```diff
@@ -20,6 +20,11 @@
 use Pimcore\Cache\RuntimeCache;
 use Pimcore\DataObject\ClassBuilder\PHPObjectBrickClassDumperInterface;
 use Pimcore\DataObject\ClassBuilder\PHPObjectBrickContainerClassDumperInterface;
+use Pimcore\Event\DataObjectClassDefinitionEvents;
+use Pimcore\Event\Model\DataObject\ClassDefinitionEvent;
```
Comment (jdreesen) [inline models/DataObject/Objectbrick/Definition.php]: Unused
```suggestion
```

File: lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Event\Model\DataObject;
+
+use Pimcore\Model\DataObject\ClassDefinition;
```
Comment (jdreesen) [inline lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php]: Unused
```suggestion
```

**Comment by BlackbitDevs** [comment]
Thank you, @jdreesen!

File: lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+use Pimcore\Model\DataObject\Objectbrick\Definition;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class ObjectbrickDefinitionEvent extends Event
+{
+    protected Definition $objectbrickDefinition;
+
+    /**
+     * @param Definition $objectbrickDefinition
+     */
```
Comment (Corepex) [inline lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php]:
```suggestion
```

File: lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    public function __construct(Definition $objectbrickDefinition)
+    {
+        $this->objectbrickDefinition = $objectbrickDefinition;
+    }
+
+    /**
+     * @return Definition
+     */
```
Comment (Corepex) [inline lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php]:
```suggestion
```

File: lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    public function getObjectbrickDefinition(): Definition
+    {
+        return $this->objectbrickDefinition;
+    }
+
+    /**
+     * @param Definition $objectbrickDefinition
+     */
```
Comment (Corepex) [inline lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php]:
```suggestion
```

File: lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    public function setObjectbrickDefinition(Definition $objectbrickDefinition): void
+    {
+        $this->objectbrickDefinition = $objectbrickDefinition;
+    }
+
+
```
Comment (Corepex) [inline lib/Event/Model/DataObject/ObjectbrickDefinitionEvent.php]:
```suggestion
```

**Review by Corepex — CHANGES_REQUESTED** [review]
LGTM

**Comment by aryaantony92** [comment]
@BlackbitDevs Thanks a lot!

**Pull request merged**